contracts/modules/common/BaseModule.sol

```diff
@@ -104,20 +112,18 @@ contract BaseModule is IModule {
      * @param _wallet The target wallet.
      * @param _module The modules to authorise.
      */
-    function addModule(address _wallet, address _module) public virtual override
-    strictOnlyWalletOwner(_wallet)
-    onlyWhenUnlocked(_wallet)
-    {
+    function addModule(address _wallet, address _module) public virtual override onlyOwner(_wallet) onlyWhenUnlocked(_wallet) {
         require(registry.isRegisteredModule(_module), "BM: module is not registered");
         IWallet(_wallet).authoriseModule(_module, true);
     }
 
      /**
-     * @dev Verify that the caller is the module or the wallet owner.
+     * @dev Verify that the caller is an authorised module or the wallet owner.
      * @param _wallet The target wallet.
+     * @param _sender The caller.
      */
-    function verifyWalletOwner(address _wallet) internal view {
-        require(msg.sender == address(this) || isOwner(_wallet, msg.sender), "BM: must be wallet owner");
+    function verifyOwnerOrModule(address _wallet, address _sender) internal view {
+        require(isModule(_wallet, _sender) || isOwner(_wallet, _sender), "BM: must be owner or module");
```

Comment:
Having an internal method reduces deployment sizes and costs, but since we have extracted all the relayer logic the modules are now much smaller. I like the idea of merging them to make the code more readable if ok with @olivdb.

Reply:
Last I checked (for MakerV2), using an internal method for `onlyWalletOwner ` instead of inlining the code directly in the modifier saved ~700k gas at deployment. I would be keen to keep `BaseModule` as lightweight as possible so it does not hinder our ability to deploy (very) large modules (ala MakerV2) in the future.

Reply:
Wow that's a lot of gas :-)

I'll keep it as it is then.